src/components/CourseForm/CourseForm.jsx

```diff
@@ -167,31 +114,116 @@ function CreateCourse(props) {
 		return true;
 	};
 
+	/**
+	 * Update the course
+	 */
+	const updateCourse = () => {
+		let durationNum = Number(duration);
+		let authors = [];
+		separateAuthorsList.course.forEach((author) => {
+			authors.push(author.id);
```

Comment:
In React we don't like to mutate the state directly. It'd be better to use setSeparateAuthorsList with whole new state - optional

Reply:
It's not a mutation, I use a new variable for array.